src/commands/create-manifest/create-manifest.ts

```diff
@@ -56,64 +65,60 @@ export const commandCreateManifest = command({
 
         // Store the list of files to move in a bucket rather than the ARGO parameters
         if (actionLocation) {
-          const targetLocation = fsa.join(actionLocation, `actions/manifest-${targetHash}.json`);
+          const targetLocation = fsa.join(actionLocation.href, `actions/manifest-${targetHash}.json`);
           const targetAction: ActionCopy = { action: 'copy', parameters: { manifest: current } };
           await fsa.write(targetLocation, JSON.stringify(targetAction));
-          outputCopy.push(targetLocation);
+          outputCopy.push(new URL(targetLocation));
         } else {
-          outputCopy.push(gzipSync(outBuf).toString('base64url'));
+          outputCopy.push(new URL(gzipSync(outBuf).toString('base64url')));
         }
       }
     }
-    await fsa.write(args.output, JSON.stringify(outputCopy));
+    await fsa.write(args.output.href, JSON.stringify(outputCopy));
   },
 });
 
-export type SourceTarget = { source: string; target: string };
+export type SourceTarget = { source: URL; target: URL };
 export type ManifestFilter = FileFilter & { flatten: boolean; transform?: string };
 
 function createTransformFunc(transform: string): (f: string) => string {
   if (transform.startsWith('return')) return new Function('f', transform) as (f: string) => string;
   return new Function('f', 'return ' + transform) as (f: string) => string;
 }
 
-export async function createManifest(
-  source: string,
-  targetPath: string,
-  args: ManifestFilter,
-): Promise<SourceTarget[][]> {
-  const outputFiles = await getFiles([source], args);
+export async function createManifest(sourceUrl: URL, targetUrl: URL, args: ManifestFilter): Promise<SourceTarget[][]> {
+  const chunks = await getFiles([sourceUrl], args);
   const outputCopy: SourceTarget[][] = [];
 
   const transformFunc = args.transform ? createTransformFunc(args.transform) : null;
 
-  for (const chunk of outputFiles) {
+  for (const chunk of chunks) {
     const current: SourceTarget[] = [];
 
-    for (const filePath of chunk) {
-      const baseFile = args.flatten ? path.basename(filePath) : filePath.slice(source.length);
-      let target = targetPath;
+    for (const chunkUrl of chunk) {
+      const baseFile = args.flatten ? path.basename(chunkUrl.href) : chunkUrl.href.slice(sourceUrl.href.length);
+      let target = targetUrl;
       if (baseFile) {
-        target = fsa.joinAll(targetPath, transformFunc ? transformFunc(baseFile) : baseFile);
+        target = new URL(fsa.joinAll(targetUrl.href, transformFunc ? transformFunc(baseFile) : baseFile));
```

Comment:
`joinAll` does a very similar job to `new URL(input, base)` in most use cases it should be a direct swap.

I believe the one caveat is `joinAll("foo.com/bar", "baz")` would make "foo.com/bar/baz" where as `new URL("baz", "foo.com/bar")` would make "foo.com/baz"

Reply:
Indeed, `new URL('sub-path', 's3://bucket/path').href === "s3://bucket/sub-path" `. But I don't know what to do about that info. I would expect most of our base URLs to be non-root ones, in which case `new URL(input, base)` would not do what we want. Which means `joinAll` is the safer alternative. Or are you saying the base URLs will always end with a slash, so that both solutions do the same thing?